### DISCUSSION TITLE
Cmd+backtick does not cycle focus to update dialog when only quick terminal window is open

### DISCUSSION THREAD

**Discussion by JakeWharton** [ghostty-org/ghostty#2892] [Feature Requests, Ideas]
I _only_ use the quick terminal, and noticed that <kbd>Cmd</kbd>+<kbd>`</kbd> will not cycle focus to the update dialog if the quick terminal is the only Ghostty window open. If I open a floating window, the quick terminal will now cycle focus to the update dialog and then to the floating window and then to the update dialog and so on... (I'm okay with it not cycling back to the quick terminal).

This is an exceedingly minor annoyance, but the asymmetry of the behavior just stood out. I would have expected either:
- <kbd>Cmd</kbd>+<kbd>`</kbd> only cycles between floating windows and never in _nor out_ of the quick terminal. Or,
- <kbd>Cmd</kbd>+<kbd>`</kbd> always cycles out of the quick terminal to either update dialogs or floating windows, but not explicitly needing a floating window to be present.

It's possible this is just a macOS quirk, and a side-effect of how the quick terminal must be implemented to work properly otherwise.

Video:

https://github.com/user-attachments/assets/e9eeb6ea-3009-4290-9828-de78bdb43dcc

**Comment by mitchellh** [comment]
I changed this to a discussion so we can figure out if this is even possible before making it an actionable issue. 

This is currently by design, caused by these lines, I'm pretty sure:

https://github.com/ghostty-org/ghostty/blob/711f94776e6a1eeb64bb5b6cfa07a262d75b42ec/macos/Sources/Features/QuickTerminal/QuickTerminalWindow.swift#L34-L45

These lines are required so that the window floats above a space so that we can place the window offscreen and animate it onto the screen. Otherwise, macOS clamps the window into the screen bounds. But, by making it not part of any space, it can't participate in cycling, specifically this:

```
// We don't want to be part of command-tilde
            .ignoresCycle,
```

I'm pretty sure all of these were required, but maybe you can play around with these and see if it fixes anything?

**Reply by JakeWharton** [reply]
So my issue is not that the quick terminal doesn't participate in cycling–that's fine. It's that the presence of a floating window changes the behavior with regard to the ability to switch to the upgrade dialog.

- From quick terminal without floating window: cannot cmd+` to update dialog
- From quick terminal with floating window: can cmd+` to update dialog
